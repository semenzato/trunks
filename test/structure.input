/* This reinforces the case that the parser can cope with comments. */

// Ditto.

typedef UINT32 foo
typedef UINT16 TPM_TAG
typedef UINT16 TPM_COMMAND_CODE

typedef struct tdBAR
{
  foo x;
  foo y;
  foo z;
} BAR;

typedef struct tdFOOBAR
{
  foo n;
  SIZEIS(n)
    BYTE *key
} FOOBAR;

