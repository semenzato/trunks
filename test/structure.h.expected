/* Copyright (c) 2011 The Chromium OS Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file.
 */

#ifndef TRUNKS_STRUCTURE_H_
#define TRUNKS_STRUCTURE_H_

#include <stdint.h>
#include <string.h>

#include "trunks_tpm.h"

typedef uint32_t foo;
typedef uint16_t TPM_TAG;
typedef uint16_t TPM_COMMAND_CODE;

typedef struct BAR {
  foo x;
  foo y;
  foo z;
} BAR;

void TtlMarshal_BAR(BAR* record, uint8_t** cursorp);
void TtlUnmarshal_BAR(BAR* record, uint8_t** cursorp);
uint32_t TtlSizeCalc_BAR(BAR* record);

typedef struct FOOBAR {
  foo n;
  uint8_t* key;
} FOOBAR;

void TtlMarshal_FOOBAR(FOOBAR* record, uint8_t** cursorp);
void TtlUnmarshal_FOOBAR(FOOBAR* record, uint8_t** cursorp);
uint32_t TtlSizeCalc_FOOBAR(FOOBAR* record);

#endif  /* TRUNKS_STRUCTURE_H_ */
